frontend/src/app/commonComponents/Checkboxes.tsx

```diff
@@ -0,0 +1,82 @@
+import React from "react";
+import classnames from "classnames";
+import useUniqueIds from "./useUniqueIds";
+
+// Checkbox objects need a value and label but also can have intrinsic `input`
+// DOM properties such as `disabled`, `readonly`, `aria-xxx` etc.
+export type CheckboxProps = {
+  value: string;
+  label: string;
+};
+type InputProps = JSX.IntrinsicElements["input"];
+type Checkbox = CheckboxProps & InputProps;
+interface Props {
+  boxes: Checkbox[];
+  checkedValues?: { [key: string]: boolean | undefined };
+  legend: React.ReactNode;
+  legendSrOnly?: boolean;
+  name: string;
+  disabled?: boolean;
+  className?: string;
+  errorMessage?: string;
+  validationStatus?: "error" | "success";
+  onChange: React.ChangeEventHandler;
+}
+
+const Checkboxes = (props: Props) => {
+  const {
+    boxes,
+    name,
+    legend,
+    checkedValues = {},
+    onChange,
+    legendSrOnly,
+    validationStatus,
+    errorMessage,
+  } = props;
+  const checkboxIds = useUniqueIds("check", boxes.length);
+
+  return (
+    <fieldset
+      className={classnames("usa-fieldset prime-checkboxes", props.className)}
+    >
+      <legend
+        className={classnames("usa-legend", legendSrOnly && "usa-sr-only")}
+      >
+        {legend}
+      </legend>
+      {validationStatus === "error" && (
```

Comment:
This is the main thing we wanted. If you pass in `validationStatus="error"` the `errorMessage` appears over the input telling you what went wrong. In the next PR we'll need to determine when that occurs, for example we probably don't want to have the error appear at load but only when the user entered something and leaves the field with an error.